### PR TITLE
Migration from google doc to markdown files

### PR DESCRIPTION
**Pull request by tthyer**
This migrates the rough draft of the developer handbook from the Google doc to this repository. The text is converted into markdown files that have a block of metadata syntax for the sake of [pelican](https://docs.getpelican.com/en/latest/content.html).

Note, although the repository is set up for pelican, it is not currently publishing pelican content automatically to gh-pages (see #1).

### CODE REVIEW

File: content/pages/000_meta/philosophy.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+Category: Meta
+Slug: philosophy
+Summary: Philosopy
+Status: published
+
+* needs work*
+
+*TODO: we should explain why we need coding conventions. Why are we bothering about this? I keep thinking in metaphors like, for coders this is how we live. It is like what you prefer to do in your own house. It really matters to some people where you put the milk in the refrigerator, or whether the utensils are put in the dishwasher pointy-side up or down.*
+
+*Alternately, some of the conventions are ways to make your work go more smoothly, or are forms of etiquette that help in collaboration.*
```
Comment (tthyer) [inline content/pages/000_meta/philosophy.md:12]: The next PR will replace the above text.

File: content/pages/003_testing/testing.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+Status: published
+
+There are many ways to test your code. The good news is that you have probably already done it many times – every time you make a change to your script/application then run it, that is testing. This, in particular, is called manual testing.
+
+Another form of testing is known as automated testing, and this relies on another script to test your code (usually done with a testing framework). You can find recommendations to some testing frameworks below under "Language-Specific Conventions" in the "Package Recommendations" sections.
+
+There are two kinds of automated testing: unit tests and integrated tests. Unlike integrated tests, which tests multiple functionalities together, a unit test will (ideally) focus on one functionality only, and prove it correct. Unit tests are useful for instances such as when you make a change to the code and the script/application breaks; your testing suite should then help pinpoint where the bug could be.
+
+*TODO: insert animated gif here*
+<figure>
```
Comment (tthyer) [inline content/pages/003_testing/testing.md:15]: Note placeholder -- I'm running out of time to do image editing.

**Review by jgockley62 — APPROVED** [review]

**Review by thomasyu888 — APPROVED** [review]

File: content/pages/001_setup/repository_creation.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    * Gitignore (hidden OS files)
+    * Code of Conduct
+    * Contribution Guidelines
+    * Dockerfile (with base alpine/ubuntu?)
+* Language-specific templates
+    * Gitignore (language-specific rules from [gitignore.io](https://www.toptal.com/developers/gitignore))
+    * Package management files (e.g., Pipfile, renv)
+    * Language-specific Dockerfile (e.g., python/rocker base image)
+
+## Repository Structure
```
Comment (philerooski) [inline content/pages/001_setup/repository_creation.md:27]: Some common use cases would be useful and encourage more consistent structures (as opposed to tossing a bunch of scripts into directories -- as I've been guilty of! Two obvious ones:

1. Python packages
2. R packages

And we can include links to tools that will auto-create necessary resources and simplify package management, like [devtools](https://www.r-project.org/nosvn/pandoc/devtools.html) for R.

File: content/pages/002_development/containerizing_applications.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+| Base Docker Image | Use Cases |
+| -|-|
+| [ubuntu](https://hub.docker.com/_/ubuntu) | larger images, but you might want to use this if you're creating a development environment |
+| [alpine](https://hub.docker.com/_/alpine) | building a tool for publishing / desiring very smallest possible footprint |
+| [python:3.x](https://hub.docker.com/_/python) | |
+| [python:3.x-slim](https://hub.docker.com/_/python?tab=tags&page=1&name=slim) | |
+| [rhub/r-minimal](https://github.com/r-hub/r-minimal) | Alpine linux-based R instance, small image size |
+| [rocker/r-base](https://hub.docker.com/r/rocker/r-base/) | R instance, usually bleeding edge when it comes to system dependencies |
+| [rocker/r-ubuntu](https://hub.docker.com/r/rocker/r-ubuntu) | Long-term support for ubuntu |
+| [rocker/r-base](https://hub.docker.com/r/rstudio/r-base) | Long-term support for ubuntu |
```
Comment (philerooski) [inline content/pages/002_development/containerizing_applications.md:25]: Let's include [tidyverse](https://hub.docker.com/r/rocker/tidyverse) images. It's so common that most use cases will prefer it to using r-base.

File: content/pages/006_languages/r.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+Summary: R conventions
+Status: published
+
+## R
+* [R](https://www.r-project.org/)
+
+## Style Guide
+* [Tidyverse style guide](https://style.tidyverse.org/index.html)
+
+## Package Recommendations
```
Comment (philerooski) [inline content/pages/006_languages/r.md:14]: Add devtools

File: content/pages/006_languages/python.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    * [pylint](https://pylint.org/)
+* HTTP requests
+    * [requests](https://docs.python-requests.org/en/latest/)
+* Math
+    * [numpy](https://numpy.org/)
+* Data
+    * [pandas](https://pandas.pydata.org/)
+    * [pyspark](https://spark.apache.org/docs/latest/api/python/)
+    * [pyarrow](https://arrow.apache.org/docs/python/install.html)
+* Visualization
```
Comment (philerooski) [inline content/pages/006_languages/python.md:38]: Add [seaborn](https://seaborn.pydata.org/index.html) and [plotly](https://plotly.com/python/)

**Review by philerooski — APPROVED** [review]

File: content/pages/002_development/containerizing_applications.md (hunk 4 above)
Comment (tthyer) [reply content/pages/002_development/containerizing_applications.md:25]: @philerooski Thanks...I will add that; this section is going to take a lot more work.

**Pull request merged**